components/chapters/SignUpModal.tsx

```diff
@@ -0,0 +1,121 @@
+'use client'
+
+import Image from 'next/image'
+import { useState } from 'react'
+import Modal from 'react-modal'
+const crypto = require('crypto')
+const Secp256k1 = require('@lionello/secp256k1-js')
+
+export const SignUpModal = (props) => {
+  function generateKeyPair() {
+    let publicKey = null;
+    let privateKey = null;
+    do {
+      const privateKeyBuf = crypto.randomBytes(32)
+      privateKey = Secp256k1.uint256(privateKeyBuf, 16)
+      publicKey = Secp256k1.generatePublicKeyFromPrivateKeyData(privateKey);
+    } while (!publicKey || privateKey.gte(Secp256k1.uint256("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141", 16)))
```

Comment:
Thanks for adding this check! I would assume that the library you are using has this covered since it's specific to secp256k1, but I think it's still a good idea to have this safety net, just to be double sure.

Reply:
I originally thought this too but after looking at the code (https://github.com/lionello/secp256k1-js/blob/master/src/secp256k1.js), the curve order doesn't seem to be checked so I added it.